src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/SourceRepositoryExtensions.cs

```diff
@@ -176,6 +176,10 @@ public static Task<SearchResult<IPackageSearchMetadata>> SearchAsync(this Source
 
             using (var sourceCacheContext = new SourceCacheContext())
             {
+                // Update http source cache context MaxAge so that it can always go online to fetch
+                // latest versions of the package.
+                sourceCacheContext.MaxAge = DateTimeOffset.UtcNow;
```

Comment:
I find this code very very confusing. 

Whether caching is utilized matters to the caller, yet the caller gets no say and should know whether method caches internally, because it's not a parameter or called out explicitly in the name.

I know it's been like this since forever, but it's a pain to review.  

Looking at the call stacks in the code, this would mean that every powershell command skips the caching as well, which I assume is ok, because powershell is only for explicit operations.